From 4726094cf5a43ad36c724ec7098dee5af37814fc Mon Sep 17 00:00:00 2001
From: Slava Pestov <sp@daterainc.com>
Date: Mon, 21 Apr 2014 18:22:35 -0700
Subject: [PATCH 03/14] bcache allocator: send discards with correct size
Git-commit: 8b326d3a2a76912dfed2f0ab937d59fae9512ca2
Patch-mainline: v3.17
References: bnc#908592

Signed-off-by: Joshua Schmid <jschmid@suse.com>

---
 drivers/md/bcache/alloc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/bcache/alloc.c b/drivers/md/bcache/alloc.c
index 443d03f..8eeab72 100644
--- a/drivers/md/bcache/alloc.c
+++ b/drivers/md/bcache/alloc.c
@@ -331,7 +331,7 @@ static int bch_allocator_thread(void *arg)
 				mutex_unlock(&ca->set->bucket_lock);
 				blkdev_issue_discard(ca->bdev,
 					bucket_to_sector(ca->set, bucket),
-					ca->sb.block_size, GFP_KERNEL, 0);
+					ca->sb.bucket_size, GFP_KERNEL, 0);
 				mutex_lock(&ca->set->bucket_lock);
 			}
 
-- 
2.1.2

